src/pages/CryptoTopUpPage.tsx

```diff
@@ -0,0 +1,170 @@
+import { useEffect, useState } from "react";
+import { ErrMsgCallbackAsProps } from "../types";
+import {
+  ardriveAppUrl,
+  maxARAmount,
+  turboConfig,
+  wincPerCredit,
+} from "../constants";
+import { Page } from "./Page";
+import { useLocation } from "react-router-dom";
+import {
+  TurboAuthenticatedClient,
+  TurboFactory,
+  ARToTokenAmount,
+  ArconnectSigner,
+} from "@ardrive/turbo-sdk";
+import { getArconnect } from "../utils/arconnect";
+import { NewToArDriveInfo } from "../components/NewToArDriveInfo";
+import { useWincForOneGiB } from "../hooks/useWincForOneGiB";
+import useDebounce from "../hooks/useDebounce";
+
+function CryptoTopUpForm({ errorCallback }: ErrMsgCallbackAsProps) {
+  const [arAmount, setArAmount] = useState<number>(0.000_000_000_200); // TODO: all allowed token types
+
+  const [turbo, setTurbo] = useState<undefined | TurboAuthenticatedClient>(
+    undefined,
+  );
+  const [sending, setSending] = useState<boolean>(false);
+
+  useEffect(() => {
+    getArconnect().then(async () => {
+      const arconnect_signer = new ArconnectSigner(window.arweaveWallet);
+      if (arconnect_signer.publicKey === undefined) {
+        await arconnect_signer.setPublicKey();
+      }
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
do we know why we need this?

Reply:
we no longer need it (:

Reply:
[e2a83ac](https://github.com/ardriveapp/turbo-app/pull/13/commits/e2a83ace674b13867e3561cf3ee6df9ff586cb57)